### DISCUSSION TITLE
Old icons instead of new ones, or a theme selector setting???

### DISCUSSION THREAD

**Discussion by wiebenor** [Ralim/IronOS#1126] [Ideas]
Was wondering if there was a way you could make the TS100 firmware have an icon theme selector, like say, a toggle, to switch between new and old menu icons, or perhaps switch individual icons, for instance, if certain new ones are liked by an individual, but they like the rest to remain the old style??? Would such a setting take up any extra disk space, and if so, would the proposed feature(s) make it too big to flash??? Or perhaps you could have an application that would be able to change the icons to any one available in a premade icon library, then apply those changes to the hex bootloader image, before copying over to the USB drive???

All in all, the new menu icons are fine, but I miss the old 3 thunderbolt power settings icon, and although the new icon for the advanced settings menu is cool, I think if I remember correctly, the old one was an animated wrench, which I liked better, because, even though the knobs and sliders on the new one are pretty informative, I just love the image of settings being a wrench, because that's what I'm used to seeing, in almost every diagnostic or advanced settings thing we as a society these days, like in vehicles, for instance.

On a related note, I forget what the solder icon looked like compared to the new one, but the old settings icon next to it was also a wrench, albeit like before, nothing is inherently wrong with the new audio mixer looking settings icon.

Of course, something else I've noticed in the repository here, is that there are many things that are added to the update notes that don't have a specific model of iron listed, or described in the update/fix notes themselves, so for someone new in the custom iron firmware scene, it might be hard to know what version is best, in my case, for the TS100  soldering iron, or what settings and improvements may have been made to that iron specifically, which in and of itself, would aid the confusion on which version to use.

One last note, I'm not myself, a developer, and I know it's probably hard to do just what you guys do here already, without all my questions and ideas that in the grand scheme, aren't important, so even if nothing I have said here is a good idea, that's ok, and I want to take a second, to thank you for making this awesome Iron Operating System, recent changes included, as it's way more intuitive than the standard miniware OS...

**Comment by Ralim** [comment]
Alrighty, to answer the sorta parts separately:

## Change icons with older ones
We cant really build this into the firmware easily as some devices have very little space left (TS100/TS80/TS80P).
This could be built for Pinecil and MHP30 but that would really be it. And building it all out for just two devices feels a bit odd.
That _said_.

## Selectable themes

This _could_ be done, if someone wanted to step up and make the tooling for the user interface side of this.
Currently there is a 1k block of flash left aside for the boot logo (which is using like 192 bytes of that).
Originally there was the idea of an animated boot logo (which could fill the rest of the block) but no one has built this out :)
What could be done instead is to fill the rest of this page with the logos; and then have the user flash their desired theme alongside the boot logo.

This would require some means for people to create the theme and extra education on how it works.


## Update notes

> Of course, something else I've noticed in the repository here, is that there are many things that are added to the update notes that don't have a specific model of iron listed, or described in the update/fix notes themselves, so for someone new in the custom iron firmware scene, it might be hard to know what version is best, in my case, for the TS100 soldering iron, or what settings and improvements may have been made to that iron specifically, which in and of itself, would aid the confusion on which version to use.

Generally, you take the one at the top. If that has issues, try the last tagged "stable" one.
Most changes apply to all or are obvious they don't (PD doesn't apply to products that don't have PD capability)

**Reply by wiebenor** [reply]
Ah, I'm sorry for having so much in one discussion post, but it looks like you answered it all, or at least, as good as I could have hoped. As for changing icons and themes, I lumped them together, because they're the same, as far as what I was referring to, but that themes section kinda covered the changing icons bit... I know there's an EXE program for converting a bootlogo, and if someone knew how, it might not be hard to make an EXE file that loads an icon library DLL, or something like that, that would allow you to choose individual icons, and preview the animation associated with it and possibly even have a "virtual soldering iron" to preview the final product on, before you export the file as a HEX file to upload where you said, next to the bootlogo... but again I'm not a developer, so I'll leave the particulars of how it would get implemented, or if other ways would work better, to others more knowledgeable

You mentioned an animated bootlogo??? That sounds cool too, if it could be fleshed out somehow
